src/metabase/util.cljc

```diff
@@ -917,3 +917,31 @@
               (map-all f (rest s1) (rest s2)))))))
   ([f c1 c2 & colls]
    (map-all* f (list* c1 c2 colls))))
+
+(defn row-diff
+  "Given two vectors of row maps - `existing` is what is in the database, `incoming` is what you want to see there,
+  returns a map of 4 lists: `{to-skip, to-delete, to-update, to-insert}`. Executing on those lists will bring database
+  to a desired state.
+
+  Additional arguments:
+  - `get-id`: function to retrieve identifier on both lists to match one to another
+  - `non-data-keys`: vector of keys which shouldn't be considered when comparing records"
```

Comment:
`non-data-keys` makes the assumption that the caller knows what keys to exclude, rather than include. In some contexts it's better to have a set of keys that we know to compare them on, rather than the keys to exclude. We do that with [dashcard updates](https://github.com/metabase/metabase/blob/b7b94edc07814bb4d555175752fa07c5c2afdc06/src/metabase/models/dashboard_card.clj#L172-L188), for example. 

If you turned it around and made `non-data-keys` a function that returned the keys which should be considered, we could use `row-diff` more places, like for dashcard updates.